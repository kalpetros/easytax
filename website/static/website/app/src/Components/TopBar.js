import styles from '../../../css/src/Components/TopBar.css';

import React from 'react';

class TopBar extends React.Component {
    constructor(props) {
        super(props);
    }
    render() {
	return(
	    <div className={styles.topBar}>
              <div className={styles.left}>
                {this.props.left}
              </div>
              <div className={styles.right}>
                {this.props.right}
              </div>
            </div>
	);
    }
}

export{TopBar};
