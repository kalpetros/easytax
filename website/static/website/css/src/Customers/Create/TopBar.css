.topBar {
    background-color: #fff;
    height: auto;
    width: 100%;
    display: grid;
    grid-auto-flow: column;
    align-items: center;
    position: sticky;
    top: 0;
    filter: drop-shadow(0 2px 2px #eee);
    z-index: 1;
}

.left {
    padding: 0.75rem;
    display: grid;
    grid-auto-flow: column;
    grid-template-columns: repeat(2, minmax(auto, max-content));
    align-items: center;
}

.right {
    padding: 0.75rem;
    display: grid;
    grid-template-columns: repeat(3, minmax(auto, max-content));
    justify-content: right;
    align-items: center;
}

.entries {
    color: #9e9e9e;
}
