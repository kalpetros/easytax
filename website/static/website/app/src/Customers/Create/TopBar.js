import styles from '../../../../css/src/Customers/Create/TopBar.css';
import {Button} from '../../Components/Button';

import React from 'react';

class TopBar extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
        };

    }
    componentDidMount() {
    }
    render() {
	return(
	    <div className={styles.topBar}>
              <div className={styles.left}>
                <Button icon="arrow_back"
                        type={"round"}
                        onClick={this.props.onBackClick}/>
              </div>
              <div className={styles.right}>
                <Button icon="save"
                        type={"round"}
                        onClick={this.props.onAddClick}/>
              </div>
            </div>
	);
    }
}

export{TopBar};
