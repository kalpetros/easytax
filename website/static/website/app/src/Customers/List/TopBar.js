import styles from '../../../../css/src/Customers/List/TopBar.css';
import {Button} from '../../Components/Button';

import React from 'react';

class TopBar extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
        };

    }
    componentDidMount() {
    }
    render() {
	return(
	    <div className={styles.topBar}>
              <div className={styles.left}>
                <Button icon="search"
                        type={"round"}/>
              </div>
              <div className={styles.right}>
                <Button icon="reorder"
                        type={"round"}/>
                <Button icon="delete_forever"
                        type={"round"}
                        onClick={this.props.onDeleteClick}/>
                <Button icon="add"
                        type={"round"}
                        onClick={this.props.onAddClick}/>
              </div>
            </div>
	);
    }
}

export{TopBar};
