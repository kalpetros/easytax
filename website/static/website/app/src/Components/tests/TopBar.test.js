import {TopBar} from '../TopBar';

import renderer from 'react-test-renderer';

import React from 'react';

test('Renders an empty top bar', () => {
  const component = renderer.create(
    <TopBar/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a top bar with content on the left side', () => {
  const component = renderer.create(
    <TopBar left="Test"/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a top bar with content on the right side', () => {
  const component = renderer.create(
    <TopBar right="Test"/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Renders a top bar with content on the left and right side', () => {
  const component = renderer.create(
      <TopBar left="Test"
              right="Test"/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
