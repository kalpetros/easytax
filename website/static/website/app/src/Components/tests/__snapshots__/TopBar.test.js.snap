// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a top bar with content on the left and right side 1`] = `
<div
  className="topBar"
>
  <div
    className="left"
  >
    Test
  </div>
  <div
    className="right"
  >
    Test
  </div>
</div>
`;

exports[`Renders a top bar with content on the left side 1`] = `
<div
  className="topBar"
>
  <div
    className="left"
  >
    Test
  </div>
  <div
    className="right"
  />
</div>
`;

exports[`Renders a top bar with content on the right side 1`] = `
<div
  className="topBar"
>
  <div
    className="left"
  />
  <div
    className="right"
  >
    Test
  </div>
</div>
`;

exports[`Renders an empty top bar 1`] = `
<div
  className="topBar"
>
  <div
    className="left"
  />
  <div
    className="right"
  />
</div>
`;
